client/cjdroute2.c

```diff
@@ -539,8 +548,78 @@ static void checkRunningInstance(struct Allocator* allocator,
     Allocator_free(alloc);
 }
 
+/* This section of code is needed so that Windows 64 can dump the stack properly
+   from a MingGW cross compiled program.  This is pretty much the only way...
+*/
+static char const * cjdroute2_global_program_name;
+#if defined(__MINGW64__)
+#pragma GCC push_options
+#pragma GCC optimize ("O0")
+#pragma GCC diagnostic ignored "-Wunused-function"
+static char const * cjdroute2_global_program_name;
+int cjdroute2_addr2line(char const * const program_name, void const * const addr);
+int cjdroute2_addr2line(char const * const program_name, void const * const addr)
+{
+  char addr2line_cmd[512] = {0};
+  /* have addr2line map the address to the relent line in the code */
+  #ifdef __APPLE__
+    /* apple does things differently... */
+    sprintf(addr2line_cmd,"atos -o %.256s %p", program_name, addr);
+  #else
+    sprintf(addr2line_cmd,"addr2line -f -p -e %.256s %p", program_name, addr);
+  #endif
+  return system(addr2line_cmd);
+}
+void cjdroute2_windows_print_stacktrace(CONTEXT* context);
+void cjdroute2_windows_print_stacktrace(CONTEXT* context)
+{
+  SymInitialize(GetCurrentProcess(), 0, true);
+  STACKFRAME frame = { 0 };
+  /* setup initial stack frame */
+  frame.AddrPC.Offset         = context->Rip;
+  frame.AddrPC.Mode           = AddrModeFlat;
+  frame.AddrStack.Offset      = context->Rsp;
+  frame.AddrStack.Mode        = AddrModeFlat;
+  frame.AddrFrame.Offset      = context->Rbp;
+  frame.AddrFrame.Mode        = AddrModeFlat;
+  while (StackWalk(IMAGE_FILE_MACHINE_AMD64 ,
+                   GetCurrentProcess(),
+                   GetCurrentThread(),
+                   &frame,
+                   context,
+                   0,
+                   SymFunctionTableAccess,
+                   SymGetModuleBase,
+                   0 ) )
+  {
+    cjdroute2_addr2line(cjdroute2_global_program_name, (void*)frame.AddrPC.Offset);
+  }
+  SymCleanup( GetCurrentProcess() );
+}
+uintptr_t __stack_chk_guard = 0xdeadbeefa55a857;
+void __stack_chk_fail(void); // CHECKFILES_IGNORE
+void __stack_chk_fail(void)  // CHECKFILES_IGNORE
+{
+#if DEBUG
+    fprintf(stderr, "cjdroute2.c:__stack_chk_guard: Stack_guard: Stack smashed\n");
+
+    CONTEXT context;
+    RtlCaptureContext(&context);
+    cjdroute2_windows_print_stacktrace(&context);
+    abort();
```

Comment:
I think you want the abort() to be outside of the if debug, also can we do an   `if (Defined(DEBUG))` ? By doing it this way we will be sure to check the code with the compiler even if debug is not enabled.